client/scripts/DrawTogether.js

```diff
@@ -1295,6 +1326,262 @@ DrawTogether.prototype.handlePaintSelection = function handlePaintSelection (eve
+		                  this.paint.public.canvas.height / this.paint.public.zoom];
+	  
+		var x = e.srcElement.parentElement.dataset.x;
+		var y = e.srcElement.parentElement.dataset.y;
+		var name = e.srcElement.parentElement.dataset.name;
+		var centerX = parseInt(this.paint.public.leftTopX + screenSize[0] / 2);
+		var centerY = parseInt(this.paint.public.leftTopY + screenSize[1] / 2);                  
+		if(e.srcElement.classList.contains("fav-button-confirmation")){
+			var coord = e.srcElement.parentElement.getElementsByClassName("fav-coor-button")[0];
+			coord.innerHTML = coord.dataset.tempConf;
+			coord.dataset.tempConf = "5215random_string_tempconf5152";
+			e.srcElement.innerHTML = "+";
+			e.srcElement.classList.remove("fav-button-confirmation");
+			
+			
+			this.setCoordFavorite(centerX, centerY, x, y, name, e.srcElement.parentElement);
+		}
+		else {
+			var coord = e.srcElement.parentElement.getElementsByClassName("fav-coor-button")[0];
+			coord.dataset.tempConf = coord.innerHTML;
+			coord.innerHTML = centerX + "," + centerY + "?";
+			e.srcElement.classList.add("fav-button-confirmation");
+			e.srcElement.innerHTML = "?";
+			setTimeout(function() {
+				var coord = this.parentElement.getElementsByClassName("fav-coor-button")[0];
+				if(coord.dataset.tempConf !== "5215random_string_tempconf5152")
+					coord.innerHTML = coord.dataset.tempConf;
+				e.srcElement.innerHTML = "+";
+				e.srcElement.classList.remove("fav-button-confirmation");
+			}.bind(e.srcElement), 2000);
+		}
+
+	}.bind(this));
+	
+	favoritePlusButton.addEventListener("mouseover", function (e) {
+		var screenSize = [this.paint.public.canvas.width / this.paint.public.zoom,
+		                  this.paint.public.canvas.height / this.paint.public.zoom];
+		
+		var centerX = parseInt(this.paint.public.leftTopX + screenSize[0] / 2);
+		var centerY = parseInt(this.paint.public.leftTopY + screenSize[1] / 2);                   
+		e.srcElement.title = "Change to " + centerX + "," + centerY + " ?";
+	}.bind(this));
+	
+	var favoritePencilButton = favoriteContainer.appendChild(document.createElement("div"));
+	favoritePencilButton.className = "fav-button fav-pencil-button";
+	favoritePencilButton.innerHTML = "✎";
+	
+	var favoriteRenameContainer = favoritePencilButton.appendChild(document.createElement("div"));
+	favoriteRenameContainer.className = "fav-rename-container";
+	
+	var favoriteRenameInput = favoriteRenameContainer.appendChild(document.createElement("input"));
+	favoriteRenameInput.className = "fav-rename-input";
+	favoriteRenameInput.type = "text";
+	favoriteRenameInput.placeholder = "Rename"
+	
+	favoriteRenameInput.addEventListener("input", function (e) {
+		if(this._favoriteRenameDelayTimeout !== undefined)
+			clearTimeout(this._favoriteRenameDelayTimeout);
+		this._favoriteRenameDelayTimeout = setTimeout(function () {
+			var newName = e.srcElement.value;
+			var greatgrandfather = e.srcElement.parentElement.parentElement.parentElement;
```

Comment:
Greatgrandfather is not a descriptive name here. Is it the current favourite? is it something else? Its hard to know from just reading this.